stub-runner/stub-runner/src/main/groovy/com/ofg/stub/StubRunnerExecutor.groovy

```diff
@@ -35,9 +35,11 @@ class StubRunnerExecutor {
     private void startStubServers(Collection<ProjectMetadata> projects, StubRepository repository) {
         projects.each { ProjectMetadata projectMetadata ->
             List<MappingDescriptor> mappings = repository.getProjectDescriptors(projectMetadata)
-            StubServer stubServer = new StubServer(portScanner.nextAvailablePort(), projectMetadata, mappings)
-            stubServer.start()
-            stubServers << stubServer
+            portScanner.tryToExecuteWithFreePort { int availablePort ->
```

Comment:
If `tryToExecuteWithFreePort` could return something, you would write:

```java
stubServers << portScanner.tryToExecuteWithFreePort { int availablePort ->
    return new StubServer(availablePort, projectMetadata, mappings).start()
}
```

Assuming `start()` returns `this`.